app/controllers/prepay_balance_controller.js

```diff
@@ -0,0 +1,1557 @@
+        console.log("prepay_balance_controller.show_product_adjustments")
+
+        return {
+                routeDetails: {
+                    controller: 'Prepay Balance',
+                    action: 'show_product_adjustments'
+                },
+                type: "voice",
+                reason: faker.lorem.sentence(),
+                amount: {
+                    amount: faker.random.number(),
+                    units: "EUR"
+                },
+                product:{
+                    id: faker.random.uuid(),
+                    href: faker.internet.url(),
+                    name: faker.name.findName()
+                },
+                id: faker.random.uuid(),
+                href: faker.internet.url(),
+                description: faker.lorem.sentence(),
+                requestor: {
+                    id: faker.random.uuid(),
+                    href: faker.internet.url(),
+                    name: faker.name.findName(),
+                    role: "user"
+                },
+                validFor: {
+                    startDateTime: new Date(),
+                    endDateTime: new Date()
+                },
+                requestedDate: new Date(),
+                bucket: {
+                    id: faker.random.uuid(),
+                    href: faker.internet.url(),
+                },
+                partyAccount: {
+                    id: faker.random.uuid(),
+                    href: faker.internet.url(),
+                    name: faker.name.findName()
+                },
+                relatedParty: [
+                    {
+                        id: faker.random.uuid(),
+                        href: faker.internet.url(),
+                        name: faker.name.findName(),
+                        role: "customer"
+                    }
+                ]
+        }
+        
+    },
+    create_product_transfers: function(request)
+    {
+        console.log("prepay_balance_controller.create_product_transfers")
+
+        return {
+                routeDetails: {
+                    controller: 'Prepay Balance',
+                    action: 'create_product_transfers'
+                },
+                type: "roaming voice",
+                channel: {
+                    id: faker.random.uuid(),
+                    href: faker.internet.url(),
+                    name: faker.name.findName()
+                },
+                targetId: "name",
+                amount: {
+                    amount: faker.random.number(),
+                    units: "EUR"
+                },
+                product:{
+                    id: faker.random.uuid(),
+                    href: faker.internet.url(),
+                    name: faker.name.findName()
+                }
+        }
+    },
+    index_product_transfers: function(request)
+    {
+        console.log("prepay_balance_controller.index_product_transfers")
+
+        return [
+            {
+                routeDetails: {
```

Comment:
missing id

Reply:
It is there(lower down), moved up for visibility